src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -1401,15 +1401,20 @@ export class BundleDataClient {
           // end block is exactly equal to the bundle block timestamp for the next bundle's start block. This way
           // there are no gaps in block timestamps between bundles.
           const endBlockForChain = Math.min(_endBlockForChain + 1, spokePoolClient.latestBlockSearched);
+          const [startTime, _endTime] = await Promise.all([
+            spokePoolClient.getTimestampForBlock(startBlockForChain),
+            spokePoolClient.getTimestampForBlock(endBlockForChain),
+          ]);
           // @dev similar to reasoning above to ensure no gaps between bundle block range timestamps and also
           // no overlap, subtract 1 from the end time.
           const endBlockDelta = endBlockForChain > startBlockForChain ? 1 : 0;
-          const [startTime, endTime] = [
-            await spokePoolClient.getTimestampForBlock(startBlockForChain),
-            (await spokePoolClient.getTimestampForBlock(endBlockForChain)) - endBlockDelta,
-          ];
+          const endTime = Math.max(0, _endTime - endBlockDelta);
```

Comment:
For some test cases, the end and start block are the same due to the `latestBlockSearched` in the test spoke pool client being lower than the bundle block range. Also in some tests, the timestamps are 0. This fix handles these cases.